src/Microsoft.OData.Edm/Csdl/Serialization/EdmModelCsdlSchemaXmlWriter.cs

```diff
@@ -504,7 +504,13 @@ internal override void WriteVocabularyAnnotationElementHeader(IEdmVocabularyAnno
             this.WriteOptionalAttribute(CsdlConstants.Attribute_Qualifier, annotation.Qualifier, EdmValueWriter.StringAsXml);
             if (isInline)
             {
-                this.WriteInlineExpression(annotation.Value);
+                if (annotation is IEdmVocabularyAnnotationWithDefault annotationWithDefault)

```

Comment:
>if (annotation is IEdmVocabularyAnnotationWithDefault annotationWithDefault) [](http://example.com/codeflow?start=16&length=76)

doesn't this result in the annotation value not being written if the annotation is not IEdmVocabularyAnnotationWithDefault? (i.e., it doesn't use default but is an older implementation so it doesn't implement the new interface)

Reply:
Very good point! The tests were fine, so I missed that, but that would be a breaking change -- I've now updated it, thanks! ✅

Reply:
Type checks and casts are expensive.  This results in one of each. Better to try a type cast and check for null.  Not quite a concise, but better performant.  Something like:

IEdmVocabularyAnnotationWithDefault annotationWithDefault = annotation as IEdmVocabularyAnnotationWithDefault;
if(annotationWithDefault == null || annotationWithDefault.UsesDefault)
{
...
}

Or, if you want to be a little more concise:

if(!(annotation is IEdmVocabularyAnnotationWithDefault annotationWithDefault) || annotationWithDefault.UsesDefault))
{
...
}